From 7e93f47e627bf851a341c930f1348f2c6bf28e1b Mon Sep 17 00:00:00 2001
From: yaravapa <yasoda.aravapalli@intel.com>
Date: Fri, 22 Oct 2021 22:00:01 +0530
Subject: [PATCH] [PATCH 8/8] Disable avx versions of sinf, cosf and expf
 functions

AVX2 implementation of these APIs is causing
CtsGraphicsTestCases failure

Change-Id: I70dd2a2cc4ce415ef34bb5e988fb0d50b3a96bdd
Tracked-On: OAm-95862
Signed-off-by: yaravapa <yasoda.aravapalli@intel.com>
---
 libm/x86_64/dynamic_function_dispatch.cpp | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/libm/x86_64/dynamic_function_dispatch.cpp b/libm/x86_64/dynamic_function_dispatch.cpp
index 3710873a9..94f6a3f1c 100644
--- a/libm/x86_64/dynamic_function_dispatch.cpp
+++ b/libm/x86_64/dynamic_function_dispatch.cpp
@@ -132,8 +132,8 @@ DEFINE_IFUNC_FOR(cos) {
 
 typedef float cosf_func(float __x);
 DEFINE_IFUNC_FOR(cosf) {
-    __builtin_cpu_init();
-    if (__builtin_cpu_supports("avx2")) RETURN_FUNC(cosf_func, cosf_avx2);
+    //__builtin_cpu_init();
+    //if (__builtin_cpu_supports("avx2")) RETURN_FUNC(cosf_func, cosf_avx2);
     RETURN_FUNC(cosf_func, cosf_generic);
 }
 
@@ -146,8 +146,8 @@ DEFINE_IFUNC_FOR(sin) {
 
 typedef float sinf_func(float __x);
 DEFINE_IFUNC_FOR(sinf) {
-    __builtin_cpu_init();
-    if (__builtin_cpu_supports("avx2")) RETURN_FUNC(sinf_func, sinf_avx2);
+    //__builtin_cpu_init();
+    //if (__builtin_cpu_supports("avx2")) RETURN_FUNC(sinf_func, sinf_avx2);
     RETURN_FUNC(sinf_func, sinf_generic);
 }
 
@@ -167,8 +167,8 @@ DEFINE_IFUNC_FOR(sinhf) {
 
 typedef float expf_func(float __x);
 DEFINE_IFUNC_FOR(expf) {
-    __builtin_cpu_init();
-    if (__builtin_cpu_supports("avx2")) RETURN_FUNC(expf_func, expf_avx2);
+    //__builtin_cpu_init();
+    //if (__builtin_cpu_supports("avx2")) RETURN_FUNC(expf_func, expf_avx2);
     RETURN_FUNC(expf_func, expf_generic);
 }
 
-- 
2.17.1

